### PR TITLE
Improve bundletool init

### PR DESCRIPTION
**Pull request by Heezay**

### CODE REVIEW

File: src/aapt2/mod.rs [hunk 1]
```diff
@@ -103,17 +103,12 @@ impl Aapt2 {
...
         return Ok(Command::new(aapt2));
     }
-    let sdk_path = {
-        let sdk_path = std::env::var("ANDROID_SDK_ROOT")
-            .ok()
-            .or_else(|| std::env::var("ANDROID_SDK_PATH").ok())
-            .or_else(|| std::env::var("ANDROID_HOME").ok());
-        PathBuf::from(sdk_path.ok_or(AndroidError::AndroidSdkNotFound)?)
-    };
+    let sdk_path = sdk_path_from_env().unwrap();
```
Comment (enfipy) [inline src/aapt2/mod.rs]: We need to get rid of `unwrap()` here.

File: src/emulator/mod.rs [hunk 2]
```diff
@@ -16,17 +16,12 @@ impl Emulator {
...
         return Ok(Command::new(emulator_tools));
     }
-    let sdk_path = {
-        let sdk_path = std::env::var("ANDROID_SDK_ROOT")
-            .ok()
-            .or_else(|| std::env::var("ANDROID_SDK_PATH").ok())
-            .or_else(|| std::env::var("ANDROID_HOME").ok());
-        PathBuf::from(sdk_path.ok_or(AndroidError::AndroidSdkNotFound)?)
-    };
+    let sdk_path = sdk_path_from_env().unwrap();
```
Comment (enfipy) [inline src/emulator/mod.rs]: And here.

File: tests/aapt2_link.rs [hunk 3]
```diff
@@ -43,13 +43,7 @@ fn test_link_files_with_aapt2_to_generate_apk() {
...
 
     // Defines path to Android SDK tools
-    let sdk_path = {
-        let sdk_path = std::env::var("ANDROID_SDK_ROOT")
-            .ok()
-            .or_else(|| std::env::var("ANDROID_SDK_PATH").ok())
-            .or_else(|| std::env::var("ANDROID_HOME").ok());
-        std::path::PathBuf::from(sdk_path.expect("Android SDK was not found"))
-    };
+    let sdk_path = sdk_path_from_env().unwrap();
```
Comment (enfipy) [inline tests/aapt2_link.rs:46]: Here too.

File: tests/build_aab.rs [hunk 4]
```diff
@@ -46,13 +46,7 @@ fn test_build_android_app_bundle() {
...
 
     // Defines path to Android SDK tools
-    let sdk_path = {
-        let sdk_path = std::env::var("ANDROID_SDK_ROOT")
-            .ok()
-            .or_else(|| std::env::var("ANDROID_SDK_PATH").ok())
-            .or_else(|| std::env::var("ANDROID_HOME").ok());
-        std::path::PathBuf::from(sdk_path.expect("Android SDK was not found"))
-    };
+    let sdk_path = sdk_path_from_env().unwrap();
```
Comment (enfipy) [inline tests/build_aab.rs:49]: Too.

File: src/bundletool/mod.rs [hunk 5]
```diff
@@ -80,13 +82,21 @@ impl Bundletool {
...
 }
 
+/// Find `bundletool` jar file in home directory then set environment variable and initialize it
 pub fn bundletool() -> Result<Command> {
-    let mut bundletool = Command::new("java");
-    bundletool.arg("-jar");
+    let bundletool = format!("bundletool-all-{}.jar", BUNDLETOOL_VERSION);
+    let env_value = dirs::home_dir()
+        .ok_or_else(|| Error::PathNotFound(PathBuf::from("$HOME")))
+        .unwrap()
```
Comment (enfipy) [inline src/bundletool/mod.rs]: Remove `unwrap()`.

File: src/bundletool/mod.rs [hunk 6]
```diff
@@ -80,13 +82,21 @@ impl Bundletool {
...
+        .join(bundletool);
+    std::env::set_var("BUNDLETOOL_PATH", env_value);
```
Comment (enfipy) [inline src/bundletool/mod.rs]: Remove this set env.

File: src/bundletool/mod.rs [hunk 7]
```diff
@@ -80,13 +82,21 @@ impl Bundletool {
     }
 }
 
+/// Find `bundletool` jar file in home directory then set environment variable and initialize it
 pub fn bundletool() -> Result<Command> {
-    let mut bundletool = Command::new("java");
-    bundletool.arg("-jar");
+    let bundletool = format!("bundletool-all-{}.jar", BUNDLETOOL_VERSION);
```
Comment (enfipy) [inline src/bundletool/mod.rs]: `BUNDLETOOL_VERSION` should be got from `ENV` with default fallback to const.

**Review by enfipy — CHANGES_REQUESTED** [review]

File: src/aapt2/mod.rs (hunk 1 above)
Comment (Heezay) [reply src/aapt2/mod.rs]: done

File: src/emulator/mod.rs (hunk 2 above)
Comment (Heezay) [reply src/emulator/mod.rs]: done

File: tests/aapt2_link.rs (hunk 3 above)
Comment (Heezay) [reply tests/aapt2_link.rs:46]: can't use question mark in test

File: tests/build_aab.rs (hunk 4 above)
Comment (Heezay) [reply tests/build_aab.rs:49]: can't use question mark in test

File: src/bundletool/mod.rs (hunk 5 above)
Comment (Heezay) [reply src/bundletool/mod.rs]: done

**Pull request merged**